pkg/backend/openshiftcluster.go

```diff
@@ -130,13 +135,19 @@ func (ocb *openShiftClusterBackend) heartbeat(log *logrus.Entry, doc *api.OpenSh
 		for {
 			_, err := ocb.db.OpenShiftClusters.Lease(doc.Key)
 			if err != nil {
+				// if lost lease - cancel backend workers
+				// and revoke lease
+				if errors.Is(err, database.ErrorLostLease) {
+					cancel()
+				}
 				log.Error(err)
 				return
 			}
 
 			select {
 			case <-t.C:
 			case <-stop:
+				cancel()
```

Comment:
I don't think this should be here.  There's always an extra operation which needs to be sneaked in after the heartbeat stops - either updating or deleting the doc.  Currently we don't pass a context to these operations (we probably should in the future).  If we do in the future, those operations would fail due to a cancelled context.  Better the `defer cancel()` above instead?

Reply:
If we do `defer cancel()` in in `handle()` maybe we should do it in that place only? This is quite sensitive "operation" so I would like to avoid having defined "cancelation" in few places so if a logic error occurs we know easier where to look. TL;DR: WE initiate it in the handle, lets just defere it there too?